File: free-platform-book/free-platform-book-api/src/main/java/top/yangwulang/platform/entity/book/ChapterContent.java
package top.yangwulang.platform.entity.book;

import org.babyfish.jimmer.Immutable;
import org.babyfish.jimmer.sql.*;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Null;

/**
 * 章节内容
 */
@Entity
@Table(name = "chapter_content")
public interface ChapterContent {
    /**
     * 书籍内容id
     */
    @Id
    String id();

    /**
     * 章节id
     */
    @Key
    @ManyToOne
    BookChapter chapter();

    /**
     * 章节内容
     */
    @Null String chapterContent();
}

File: platform/src/main/kotlin/top/yangwulang/platform/entity/book/ChapterContent.kt
package top.yangwulang.platform.entity.book

import org.hibernate.annotations.DynamicInsert
import org.hibernate.annotations.DynamicUpdate
import org.hibernate.annotations.GenericGenerator
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "chapter_content")
@DynamicUpdate
@DynamicInsert
open class ChapterContent : Serializable {
    @Id
    @Column(name = "id", nullable = false, length = 64)
    @GeneratedValue(generator = "snowflake")
    @GenericGenerator(name = "snowflake", strategy = "top.yangwulang.platform.utils.SnowflakeGenerator")
    open var id: String? = null

    @OneToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "chapter_id", nullable = false)
    open var chapter: BookChapter? = null

    @Lob
    @Column(name = "chapter_content")
    open var chapterContent: String? = null
}